mailstore-connector/src/com/axonivy/connector/mailstore/MailStoreService.java

```diff
@@ -609,25 +622,30 @@ public static Store openStore(String storeName) throws MessagingException {
 		String host = getVar(storeName, HOST_VAR);
 		String portString = getVar(storeName, PORT_VAR);
 		String user = getVar(storeName, USER_VAR);
-		String password = getVar(storeName, PASSWORD_VAR);
+		
+		UserPasswordProvider userPasswordProvider = userPasswordProviderRegister.get(storeName);
+		// adapt exist project already use this connector, default is basic auth
+		if(null == userPasswordProvider) {
+			userPasswordProvider = new BasicUserPasswordProvider();
+		}
+		String password = userPasswordProvider.authenticate(storeName);

```

Comment:
Other services might work differently and probably provide a user and a password. I would suggest to get user and password from the userPasswordProvider and call the methods `getUser()` and `getPassword()` to be more general.

Reply:
Hi @peters-axon 
thanks for your hint, I did refractor it.